Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1788 @@
+
+unzip -o fits-1.1.1.zip && rm -rf fits-1.1.1.zip && ln -s fits-1.1.1 fits && chmod a+x /usr/local/fits/fits.sh
+```
+
+#### Adore-Djatoka Install<a id="adore-djatoka-install"></a>
+``` 
+cd /usr/local
+
+wget http://sourceforge.net/projects/djatoka/files/djatoka/1.1/adore-djatoka-1.1.tar.gz/download -O adore-djatoka-1.1.tar.gz
+
+tar xf adore-djatoka-1.1.tar.gz && rm -rf adore-djatoka-1.1.tar.gz
+
+ln -s adore-djatoka-1.1 adore-djatoka
+```
+
+Make kakadu available on the command line:  
+```
+ln -s /usr/local/adore-djatoka/bin/Linux-x86-64/kdu_expand /usr/bin/kdu_expand
+
+ln -s /usr/local/adore-djatoka/bin/Linux-x86-64/kdu_compress /usr/bin/kdu_compress
+
+echo "/usr/local/adore-djatoka/lib/Linux-x86-64" > /etc/ld.so.conf.d/kakadu.conf
+
+ldconfig
+```
+
+#### Cantaloupe Install<a id="cantaloupe-install"></a>
+
+This is optional. Cantaloupe is an alternative large image display processer to Adore-Djatoka, for use with the Openseadragon viewer. Instructions are closely based on http://dev.digibess.it/doku.php?id=reloaded:is_cantsa, with the latest version of Cantaloupe.
+
+```
+cd /usr/local
+
+wget https://github.com/medusa-project/cantaloupe/releases/download/v3.4.2/Cantaloupe-3.4.2.zip
+
+unzip Cantaloupe-3.4.2.zip
+
+rm -f Cantaloupe-3.4.2.zip
+
+ln -s /usr/local/Cantaloupe-3.4.2 /usr/local/cantaloupe
+
+mkdir -p /srv/cantaloupe/cache
+
+mkdir /srv/cantaloupe/log
+
+mkdir /srv/cantaloupe/home
+
+mkdir /srv/cantaloupe/tmp
+```
+
+#### Drush  <a id="drush"></a>
+
+Ubuntu 16.4 has Drush version 5.10, which is installed above. If that causes issues, do the following.
+
+Specify drush commit due to issues with newer drush versions and automated tests used by QA:
+```
+cd /usr/local
+
+git clone https://github.com/drush-ops/drush.git && cd drush && git checkout b9e6c8c00da0fbf1227869cdf915b0c6cea466cc
+
+ln -s /usr/local/drush/drush /usr/bin/drush
+```
+
+### Configuration <a id="configuration"></a>  
+
+#### LibreOffice <a id="libreoffice"></a>
+
+We use Systemd, which manages restarting on failure, so Monit is no longer needed.
+```
+useradd -m -d /home/libreoffice libreoffice
+
+cat > /etc/systemd/system/libreoffice.service <<END_LO
+[Unit]
+Description=Headless Libreoffice API server
+
+[Service]
+Type=simple
+User=libreoffice
+UMask=0002
+ExecStart=/usr/bin/soffice --headless --nologo --accept="socket,host=127.0.0.1,port=8100;urp"
+ExecStop=/usr/bin/killall -9 soffice.bin
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
+END_LO
+
+systemctl daemon-reload
+
+systemctl enable libreoffice.service
+
+systemctl start libreoffice.service
+```
+
+Verify functionality with:
+
+`unoconv --connection 'socket,host=127.0.0.1,port=8100;urp;StarOffice.ComponentContext' -f pdf <some doc in a location writeable by user libreoffice>`
+
+The following steps are required so that the `apache2` user can delete derivatives created by `libreoffice` in `/tmp`.
+
+```
+adduser www-data libreoffice
+
+mkdir /tmp/fedora
```

Comment:
If it is for LibreOffice why is the temp called "/tmp/fedora"?

Reply:
Good question. I think, since this ends up being the default location
for Drupal temp material, maybe it could be 'drupal' instead of
'fedora'. But it does get used for other derivative generation, for example,
besides 'libreoffice'.